interfaces/builtin/optical_drive.go

```diff
@@ -46,13 +46,18 @@ const opticalDriveConnectedPlugAppArmor = `
 # Allow read access to optical drives
 /dev/sr[0-9]* r,
 /dev/scd[0-9]* r,
+/dev/sg[0-9]* r,
 @{PROC}/sys/dev/cdrom/info r,
 /run/udev/data/b11:[0-9]* r,
 `
 
 var opticalDriveConnectedPlugUDev = []string{
 	`KERNEL=="sr[0-9]*"`,
 	`KERNEL=="scd[0-9]*"`,
+	// ATTRS{type} below takes scsi peripheral device types.
+	// Type 4 is 'Write-once device'; type 5 is 'CD/DVD-ROM device'
+	// ref: https://en.wikipedia.org/wiki/SCSI_Peripheral_Device_Type
+	`SUBSYSTEM=="scsi_generic", SUBSYSTEMS=="scsi", ATTRS{type}=="4|5"`,
```

Comment:
This is quite nice. As for getting rid of type 4, I think it is ok to leave. Type 4 is defined as 'write-once, read many' (aka WORM) and there are historical cases of optical WORM devices (eg, https://www.computerhistory.org/collections/catalog/102671161), though the original WORM devices were punch cards. :)